src/core/services/IronBankService.ts

```diff
@@ -94,31 +94,48 @@ export class IronBankServiceImpl implements IronBankService {
     const { MAX_UINT256 } = this.config;
     const provider = this.web3Provider.getSigner();
     const ironBankMarketContract = getContract(marketAddress, ironBankMarketAbi, provider);
-
     switch (action) {
       case 'supply':
-        return await this.transactionService.execute({ network, fn: ironBankMarketContract.mint, args: [amount] });
+        return await this.transactionService.execute({
```

Comment:
what do you think about having different methods for each one of this switch cases?

executeTransaction ends up being very generic, if you split it up in methods:
- You can get advantage of TS typing by making sure you are always sending the correct parameters
- You make methods here more atomic and thus more testable

Reply:
Do agree on that approach (and we pretty much use it all over the repo) but since this piece of code is not gonna be on the FE for too long (since we are gonna move this logic to the SDK) we went for the simplest/generic approach in here.